learning_journal/models.py

```diff
@@ -32,6 +32,19 @@ class MyModel(Base):
 
 Index('my_index', MyModel.name, unique=True, mysql_length=255)
 
+class User(Base):
+    __tablename__ = 'users'
+    id = Column(Integer, primary_key=True)
+    username = Column(Unicode(255), nullable=False, unique=True, index=True)
+    password = Column(Unicode(255), nullable=False)
+
+    @classmethod
+    def by_username(cls, username, session=None):
+        if session is None:
+            session = DBSession
+        query = session.query(cls).filter(cls.username == username)
+        return query.one()  # not sure if this is the best way to do this
```

Comment:
You have a couple of choices here.  You can use `query.one()`, `query.one_or_none()`, or `query.first()`.

The differences between each of these are entirely in how they deal with errors.  Read the information about each method on [this page about the SQLAlchemy query API](http://docs.sqlalchemy.org/en/latest/orm/query.html) and make sure you are aware of the implications of your choice.

So long as you are aware of the possible errors, and are handling them, then it's a good choice.